panoply/error_normalization.py

```diff
@@ -0,0 +1,97 @@
+from panoply import DataSource, PanoplyException
+from datetime import datetime
+
+
+ERROR_CODES_REGISTRY = {
+    400: 'Bad request',
+    401: 'Unauthorized',
+    403: 'Permissions error',
+    404: 'Resource is not found',
+    408: 'Timeout',
+    422: 'Parsing error',
+    429: 'Rate limit',
+    500: 'Internal exception',
+    501: 'Unclassified error',
+    506: 'SDK error in processing an error'
+}
+
+EXCEPTIONS_REGISTRY = {}
+
+
+class NormalizedException(PanoplyException):
+    def __init__(self, message, code, exception_cls,
+                 phase, source_type, source_id, database_id,
+                 retryable=True):
+        super().__init__(message, retryable)
+        self.message = message
+        self.code = code
+        self.phase = phase
+        self.source_type = source_type
+        self.source_id = source_id
+        self.database_id = database_id
+        self.exception_cls = exception_cls
+        self.created_at = datetime.utcnow()
+
+
+def set_internal_code(code: int) -> callable:
+    """ A decorator is used on exception classes to register mapping
+        of error and internal code from ERROR_CODES_REGISTRY.
+        Parameters
+        ----------
+        code : int
+           Error code from ERROR_CODES_REGISTRY.
+    """
+    def decorator(exception_cls: type) -> type:
+        if not issubclass(exception_cls, BaseException):
+            raise RuntimeError('Unable to set error code for non-Exception class')
+
+        EXCEPTIONS_REGISTRY[exception_cls] = code
+
+        return exception_cls
+
+    return decorator
+
+
+def handle_errors(phase: str) -> callable:
```

Comment:
This decorator is not "handling" the error it just wraps it.
```suggestion
def wrap_errors(phase: str) -> callable:
```